### PR TITLE
Introduce the Code-Quality Syllabus

### PR DESCRIPTION
**Pull request by EdDev**
List the topics which are to be covered in the code-quality course.

### CODE REVIEW

File: syllabus.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  - [ ] Single responsibility principle
+  - [ ] Open-Closed principle
+  - [ ] Liskov Substitution principle
+  - [ ] Interface segregation principle
+  - [ ] Dependency inversion principle
+- [ ] Immutable functions
+- [ ] Naming and comments
+- [ ] Code Structure
+- [ ] Testing
+  - [ ] Test doubles
```
Comment (EdDev) [inline syllabus.md]: Done

File: syllabus.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- [x] The build pattern
+- [ ] Anonymous function (lambda)
+  - Closure
+- [ ] The SOLID principles
+  - [ ] Single responsibility principle
+  - [ ] Open-Closed principle
+  - [ ] Liskov Substitution principle
+  - [ ] Interface segregation principle
+  - [ ] Dependency inversion principle
+- [ ] Immutable functions
```
Comment (orelmisan) [inline syllabus.md:15]: Could you please elaborate on the meaning of "Immutable functions"?

File: syllabus.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] Naming and comments
+- [ ] Code Structure
+- [ ] Testing
+  - [ ] Test doubles
+  - [ ] Tips for testable code part 1
+  - [ ] Tips for testable code part 2
+  - [ ] Unit, Integration or Functional?
+  - [ ] Test Driven Development
+- [ ] Source control
```
Comment (orelmisan) [inline syllabus.md:24]: Please consider adding a lecture on composing PRs, and strategies for moving forward "controversial" PRs.

File: syllabus.md [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+- [ ] Anonymous function (lambda)
+  - Closure
+- [ ] The SOLID principles
+  - [ ] Single responsibility principle
+  - [ ] Open-Closed principle
+  - [ ] Liskov Substitution principle
+  - [ ] Interface segregation principle
+  - [ ] Dependency inversion principle
+- [ ] Immutable functions
+- [ ] Naming and comments
```
Comment (orelmisan) [inline syllabus.md:16]: Does that includes package naming?

File: syllabus.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  - [ ] Test doubles
+  - [ ] Tips for testable code part 1
+  - [ ] Tips for testable code part 2
+  - [ ] Unit, Integration or Functional?
+  - [ ] Test Driven Development
+- [ ] Source control
+  - [ ] The commit scope
+  - [ ] Keep the history clean (rebase)
+  - [ ] Merge and split a commit
+- [ ] CI/CD
```
Comment (orelmisan) [inline syllabus.md:28]: Please consider discussing the importance of a (quick) feedback loop.

File: syllabus.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
+# Syllabus
+
+The following topics relate to code quality.
+
```
Comment (orelmisan) [inline syllabus.md:4]: Common design patterns (in Go).

File: syllabus.md (hunk 2 above)
Comment (EdDev) [reply syllabus.md:15]: Functions that do not change the input.

File: syllabus.md (hunk 6 above)
Comment (EdDev) [reply syllabus.md:4]: The patterns I find interesting are listed below, I am not sure what you mean by this comment.

File: syllabus.md (hunk 4 above)
Comment (EdDev) [reply syllabus.md:16]: Yes

File: syllabus.md (hunk 3 above)
Comment (EdDev) [reply syllabus.md:24]: Good idea, I will add it.

File: syllabus.md (hunk 5 above)
Comment (EdDev) [reply syllabus.md:28]: I think it should be part of any talk about unit tests and CI/CD.
I do not see how it can be its own topic/lecture.

**Review by EdDev — COMMENTED** [review]
Thank you for the review.

File: syllabus.md (hunk 3 above)
Comment (EdDev) [reply syllabus.md:24]: Done

**Review by orelmisan — APPROVED** [review]
Thank you

**Pull request merged**